napi_js_props.md

```diff
@@ -0,0 +1,438 @@
+won't be used). The given function is called implicitly by the runtime when the
+property is set from JavaScript code (or if a set on the property is 
+performed using a N-API call).
+- `method`: Set this if you want the property descriptor object's `value`
+property to be a JavaScript function represented by `method`. If this is
+passed in, set `value`, `getter` and `setter` to `NULL` (since these members
+won't be used)
+- `data`: The callback data passed into `method`, `getter` and `setter` if 
+this function is invoked
+- `attributes`: The attributes associated with the particular property. 
+See [`napi_property_attributes`](#napi_property_attributes)
+
+## Functions
+
+### *napi_get_property_names*
+
+#### Signature
+```C
+napi_status napi_get_property_names(napi_env env,
+                                    napi_value object,
+                                    napi_value* result);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API call is invoked under
+- `[in]  object`: The object from which to retrieve the properties
+- `[out] result`: A `napi_value` representing an array of JavaScript values 
+that represent the property names of the object. You can iterate over `result`
+using `napi_get_array_length` and `napi_get_element`.
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_set_property*
+
+#### Signature
+```C
+napi_status napi_set_property(napi_env env,
+                              napi_value object,
+                              napi_value key,
+                              napi_value value);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API call is invoked under
+- `[in]  object`: The object on which to set the property
+- `[in]  key`: The name of the property to set
+- `[in]  value`: The property value
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_get_property*
+
+#### Signature
+```C
+napi_status napi_get_property(napi_env env,
+                              napi_value object,
+                              napi_value key,
+                              napi_value* result);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API call is invoked under
+- `[in]  object`: The object from which to retrieve the property
+- `[in]  key`: The name of the property to retrieve
+- `[out] result`: The value of the property
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_has_property*
+
+#### Signature
+```C
+napi_status napi_has_property(napi_env env,
+                              napi_value object,
+                              napi_value key,
+                              bool* result);
+```
+
+#### Parameters
+- `[in]  env`: The environment that the N-API call is invoked under
+- `[in]  object`: The object to query
+- `[in]  key`: The name of the property whose existence to check
+- `[out] result`: Whether the property exists on the object or not
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
```

Comment:
For consistency, these methods ought to have a `### Description` section, althought they *are* pretty self-explanatory.